src/calendar.cpp

```diff
@@ -257,11 +257,33 @@ bool calendar::is_night() const
     return (seconds > sunset_seconds + TWILIGHT_SECONDS || seconds < sunrise_seconds);
 }
 
+double calendar::current_daylight_level() const
+{
+    double current_daylight_modifier = 1.;
+    
+    switch (season) {
+    case SPRING:
+        current_daylight_modifier = 1.;
+        break;
```

Comment:
Way too verbose. The whole function could be shortened:
```c++
double calendar::daylight_modifier()
{
    switch (season) {
    case SUMMER:
        return 1.03;
    case WINTER:
        return 0.97;
    }
    return 1.0;
}

...

double daylight_level = double(DAYLIGHT_LEVEL) * daylight_modifier();

```
Also, smooth changes would be better than sudden ones.